apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/BooleanExpression.kt

```diff
@@ -135,3 +135,14 @@ data class BPossibleTypes(val possibleTypes: Set<String>) : BTerm() {
   constructor(vararg types: String): this(types.toSet())
 }
 
+
+fun <T : Any> BooleanExpression<T>.containsPossibleTypes(): Boolean {
+  return when (this) {
+    BooleanExpression.True -> false
+    BooleanExpression.False -> false
+    is BooleanExpression.Not -> !operand.containsPossibleTypes()
```

Comment:
Should it be `operand.containsPossibleTypes()` 🤔 To check that a type is _not_ in the list you need the type?